src/actions/wallet/L1/writeContractDeposit.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  type Abi,
+  type Account,
+  type Chain,
+  encodeFunctionData,
+  type EncodeFunctionDataParameters,
+  type Transport,
+  type WalletClient,
+  type WriteContractParameters,
+  type WriteContractReturnType,
+} from 'viem'
+import { OpStackL1Contract } from '../../../index.js'
+import type { GetL2Chain, L1ActionBaseType, ResolveChain } from '../../../types/l1Actions.js'
+import { writeSendMessage, type WriteSendMessageParameters } from './writeSendMessage.js'
+
+export type WriteContractDepositParameters<
+  TAbi extends Abi | readonly unknown[] = Abi,
+  TFunctionName extends string = string,
+  TChain extends Chain | undefined = Chain,
+  TAccount extends Account | undefined = Account | undefined,
+  TChainOverride extends Chain | undefined = Chain | undefined,
+> =
+  & { minGasLimit: number }
+  & L1ActionBaseType<GetL2Chain<ResolveChain<TChain, TChainOverride>>, typeof OpStackL1Contract.L1CrossDomainMessenger>
+  & WriteContractParameters<
+    TAbi,
+    TFunctionName,
+    TChain,
+    TAccount,
+    TChainOverride
+  >
+
+/**
+ * A L1 -> L2 version of Viem's writeContract. Can be used to call any L2 contract from L1.
+ * Internally uses writeSendMessage, which calls to the cross chain messenger contract.
+ * NOTE: msg.sender on the call to `address` will NOT be msg.sender on L1, but the L2CrossDomainMessenger.
```

Comment:
want to add #92

Reply:
Not totally happy with this naming given that nuances, thinking about like writeContractSendMessage or something @jxom @roninjin10

Reply:
This isn't true you should delete this

Reply:
Being able to send an l2 transaction via a deposit transaction is part of the security properties of the L2. Without that, the sequencer could censor you

Reply:
@roninjin10 this is true. Look at the L2 from address when sending a message. It's the alias of the l1 cross chain messenger. You cannot make L2 calls that require your L1 address as msg.sender using sendMessage.